Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,562 @@
+            options.Context.RequestContainer)).SelectExpandClause;
+
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+When making the request, we don't need to use the `$expand` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+Below will be the output:
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Orders())",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31,
+            "Orders": [
+                {
+                    "Id": 1001,
+                    "Price": 10,
+                    "Quantity": 10
+                },
+                {
+                    "Id": 1002,
+                    "Price": 35,
+                    "Quantity": 2
+                }
+            ]
+        },
+        ...
+        ...
+    ]
+}
+```
+
+### Filter query
+We call `ApplyTo` on the `Filter` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+The `options.Filter` property is an instance of `FilterQueryOption`. It will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$filter=Id eq 1
+```
+
+If we don't want to pass the filter query option in the request, but we want a filter to be applied, we can initialize the `FilterQueryOption` and call `ApplyTo` to apply the filter query to the data.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    var filter = new FilterQueryOption("Id eq 1", options.Context,
+        new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$filter", "Id eq 1" } },
+            options.Context.RequestContainer));
+
+    IQueryable results = filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+When making the request, we don't need to use the `$filter` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+Below will be the output:
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31
+        }
+    ]
+}
+```
+
+### Apply query
+Apply query feature is applied in grouping and aggregating data.
+
+We call `ApplyTo` on the `Apply` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Apply.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
```

Comment:
Refer to my other comment on overload of `ApplyTo` that doesn't expect an `ODataQuerySettings` argument

Reply:
The overload is not in `ApplyQueryOption`